### PR TITLE
ADEN-2812 IVW3 integration

### PR DESCRIPTION
**Pull request by fraszczakszymon**
Implementation of new IVW3 tracking provider which will be enabled in DE, AT, CH

### CODE REVIEW

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		$vertical = HubService::getVerticalNameForComscore( $wgCityId );
+		$url = self::$libraryUrl;
+		$iamData = json_encode( [
+				'st' => 'wikia',
+				'cp' => $vertical,
+				'sv' => 'ke',
+		] );
+
+		$ivwScriptTag = <<<CODE
+<script type="text/javascript" src="{$url}"></script>
```
Comment (nandy-andy) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php]: Can't we just use here `self::$libraryUrl`?

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+<script type="text/javascript">window.iom.c({$iamData}, 2);</script>
+CODE;
+		$ivwScriptTagEscaped = json_encode($ivwScriptTag);
+
+		return <<<CODE
+<!-- Begin IVW3 Tag -->
+<script type="text/javascript">
+	if (window.Wikia && window.Wikia.geo.isProperGeo(['DE', 'AT', 'CH'])) {
```
Comment (nandy-andy) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php]: Now I'm thinking about not an instant global but just a regular WF variable which we can use here in PHP and re-use in Mercury. But can we use it in Mercury? Like get it on first view where whole Ember is loading and Mercury asks for WF config? Can we return it with this config if it gets there before sending request to IVW? If yes, let's do it! If no, let's not waste our time and use it to think how make Mercury more flexible ;-)

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW2.php [hunk 3]
```diff
@@ -38,15 +38,17 @@ function trackEvent($event, $eventDetails = array()) {
 					"sv" => "ke"      // FRABO-Tag deaktiviert
 				];
 
-				$ivwScriptTag = '<script src="https://script.ioam.de/iam.js"></script>';
-				$ivwScriptTag .= '<script>iom.c(' . json_encode($iamData) . ', 2);</script>';
-				$ivwScriptTagEscaped = json_encode($ivwScriptTag);
+				$ivwScriptTag = json_encode('<script src="https://script.ioam.de/iam.js"></script>');
+				$ivwTrackTag = json_encode('<script>iom.c(' . json_encode($iamData) . ', 2);</script>');
```
Comment (nandy-andy) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW2.php:42]: Can you contact IVW if it hurts them if we execute it twice? However, those are different data, aren't they?

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php (hunk 1 above)
Comment (fraszczakszymon) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php]: Can't use it in HEREDOC :(

Comment (hakubo) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php]: can we not write JavaScript in php? 
You could put that at least in mustache template and render it, that alone would already be like we lived in 2016 :)

Comment (fraszczakszymon) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php]: Right! :gem: now it's much cleaner, thanks

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php (hunk 2 above)
Comment (fraszczakszymon) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php]: :+1:

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW2.php (hunk 3 above)
Comment (fraszczakszymon) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW2.php:42]: :+1: I'll send an e-mail

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php (hunk 1 above)
Comment (nandy-andy) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW3.php]: @hakubo, oh you fashion boy! :japanese_ogre: :trollface: :dash: 

Thanks for the hint! We got into "being consistent" trap. We'll try to remember it next time we touch trackers :bowtie:

File: extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW2.php (hunk 3 above)
Comment (nandy-andy) [inline extensions/wikia/AnalyticsEngine/AnalyticsProviderIVW2.php:42]: Thank you :bow:

**Comment by jacekjursza** [comment]
LGTM

**Comment by jacekjursza** [comment]
LGTM

**Pull request merged**